src/odemis/driver/tescan.py

```diff
@@ -1443,6 +1443,8 @@ def _doMoveAbs(self, pos: dict):
                     # convert from radians to degrees for the rotational axes
                     elif axis in {"rz", "rx"}:
                         req_pos[axis] = math.degrees(pos[axis])
+                else:
+                    current_pos[axis] = None
 
             current_pos.update(req_pos)
```

Comment:
Yes, this works, but now that we only want to move the axes requested, the whole code can be much simpler. We don't need to read the current position. You can directly use `req_pos` as argument to `StgMoveTo()`.

```suggestion
req_pos = {}

for axis in {"x", "y", "z", "rz", "rx"}:
    if axis in pos:
         # convert from m to mm and invert for the linear axes
         if axis in {"x", "y", "z"}:
              req_pos[axis] = -pos[axis] * 1e3
         # convert from radians to degrees for the rotational axes
         elif axis in {"rz", "rx"}:
               req_pos[axis] = math.degrees(pos[axis])
    else:
         req_pos[axis] = None
```